### PR TITLE
distros/photon.py: refactor hostname handling

### PR DESCRIPTION
**Pull request by sshedi**
Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>

### CODE REVIEW

**Comment by sshedi** [comment]
Hi @TheRealFalcon - can you please review this change? You suggested to use this in my very first PR, I took a part of hostname related chnages from RHEL distro code. Now I have tried to make it better.

File: cloudinit/distros/photon.py [hunk 1]
```diff
@@ -99,26 +98,23 @@ def _bring_up_interfaces(self, device_names):
...
-            pass
-        if not conf:
-            conf = HostnameConf('')
-        conf.set_hostname(hostname)
-        util.write_file(filename, str(conf), mode=0o644)
+        if filename.endswith('/previous-hostname'):
+            util.write_file(filename, hostname)
+        else:
+            _ret, _out, _err = self.exec_cmd(['hostnamectl', 'set-hostname',
+                                             str(hostname)])
```
Comment (TheRealFalcon) [inline cloudinit/distros/photon.py:105]:
```suggestion
            self.exec_cmd(['hostnamectl', 'set-hostname', str(hostname)])
```
If we're not using the results, then we shouldn't store them.

File: cloudinit/distros/photon.py [hunk 2]
```diff
@@ -99,26 +98,23 @@ def _bring_up_interfaces(self, device_names):
...
 
     def _read_hostname(self, filename, default=None):
-        _ret, out, _err = self.exec_cmd(['hostname'])
+        if filename.endswith('/previous-hostname'):
+            return util.load_file(filename).strip()
 
+        _ret, out, _err = self.exec_cmd(['hostname', '-f'], capture=True)
+        if out:
+            out = out.strip()
         return out if out else default
```
Comment (TheRealFalcon) [inline cloudinit/distros/photon.py:118]:
```suggestion
        return out.strip() if out else default
```
Just a minor tweak.

**Review by TheRealFalcon — APPROVED** [review]
Looks good to me!

I added two small minor suggestions inline, but they won't block merge.

File: cloudinit/distros/photon.py (hunk 1 above)
Comment (sshedi) [reply cloudinit/distros/photon.py:105]: Agree.

File: cloudinit/distros/photon.py (hunk 2 above)
Comment (sshedi) [reply cloudinit/distros/photon.py:118]: Thanks.

**Comment by sshedi** [comment]
@TheRealFalcon - I have added networkd activator in this PR now. Please take a look.
If you want me to send a different PR for network activator changes, I will do so.

**Review by TheRealFalcon — APPROVED** [review]
Looks good, thanks!

**Comment by sshedi** [comment]
Welcome and thank you too.

**Pull request merged**